web/src/people.js

```diff
@@ -1198,8 +1198,18 @@ export function get_mention_syntax(full_name, user_id, silent) {
     return mention;
 }
 
-function full_name_matches_wildcard_mention(full_name) {
-    return ["all", "everyone", "stream"].includes(full_name);
+function extract_name_from_mention(mention) {
+    if (!mention.startsWith('@') || !mention.endsWith('**')) {
+        return mention;
+    }
+    const start_index = mention.indexOf('_**') === 1 ? 4 : 3;
+
+    return mention.slice(start_index, -2);
+}
+
+export function is_wildcard_mention(mention) {
+    const name = extract_name_from_mention(mention);
+    return ["all", "everyone", "stream"].includes(name);
 }
```

Comment:
Do callers know whether they're expecting to have a name or mention syntax? I would usually expect so, and that's the case for `get_mention_syntax` above.

For that reason I think it'd be better to have two functions; `is_wildcard_mention` and `is_wildcard_mention_syntax`, with the latter requiring something starting with `@`, and the former requiring it not to.

Reply:
The callers are aware of what they're passing, yeah.

I was wondering, do you think it would be a good idea to raise a blueslip error or a warning if someone accidentally passes something starting with '@' to the wrong function? I'm not sure how often this will come up, but it might be worth considering just to be on the safe side. What do you think?

Reply:
This changes was rolled back.